### ISSUE TITLE
twisted.test.test_application.TestInternet2.testConnectionGettingRefused is racey

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2480]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2480|
|Type|defect|
|Created|2007-02-25 16:00:31Z|

This test sets up and then tears down a server, then attempts to connect to it with a client to assert that the correct behavior is provided by TCPClient when a connection attempt fails.

The test does not wait for the server to stop listening before attempting to connect, though.  This means the server might still be available when the client connection attempt is made.

TCPServer.stopService returns a Deferred.  The test should wait on it before attempting a connection.

Alternatively, real sockets shouldn't be used at all here.

<details><summary>Searchable metadata</summary>

```
trac-id__2480 2480
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__new new
resolution__None None
component__core core
keywords__ 
time__1172419231000000 1172419231000000
changetime__1172419231000000 1172419231000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
